Makefile

```diff
@@ -1,12 +1,44 @@
+PUB_DIR := ./pub
+
 .PHONY: lint
 lint:
 	ruff check --exit-zero .
+	ruff format --check .
 
 .PHONY: format
 format:
 	ruff check --fix .
 	ruff format .
 
+.PHONY: typecheck
+typecheck:
+	pyright --project pyproject.toml .
+
 .PHONY: pre-commit
 pre-commit:
 	pre-commit run --all-files
+
+.PHONY: test
+test:
+	pytest -v .
+
+# By default, `run-notebook` will just execute notebook.ipynb (overwriting
+# current state of notebook) using `jupyter nbconvert`. If the analysis
+# includes additional setup, `run-notebook` should be updated to include these
+# steps.
+.PHONY: run-notebook
+run-notebook:
```

Comment:
Yeah this is a fair question.

I reworked things in 10b731f753214e7eb7ce56c1a8ae91c905030d90 so that `make run-notebook` is externally facing. I did this by eliminating the need to deploy a permanent modal app as a setup step. Instead, ephemeral apps are created as needed. This is better anyways because you can treat your local as the head node instead of spawning a remote head node.

I don't want to force a potentially expensive re-run every time `make pub` is called, but I've better documented `make run-notebook` and why you want to use it.

Reply:
Nice!